app/policies/application_policy.rb

```diff
@@ -62,7 +62,12 @@ def can_admin?
   end
 
   def own_record?
-    user.present? && record.user == user
+
+    user.present? && 
+    (
+      record == user || 
+      (record.respond_to?(:user) && record.user == user)
```

Comment:
Consider using Rails' `Object#try` method http://api.rubyonrails.org/classes/Object.html#method-i-try 

Also consider introducing an `owner` method to wrap some of this logic.